### PR TITLE
Fix docker committing and error handling.

### PR DESCRIPTION
**Pull request by LinoBigatti**
The way the program handled docker commits and error codes was flawed and prevented correct functioning. This fixes these issues. Related issue: #73

### CODE REVIEW

File: etest/docker/container.py [hunk 1]
```diff
@@ -48,13 +50,13 @@ def logs(*args, **kwargs):
     return common.API_CLIENT.logs(*args, **kwargs)
 
 
-def remove(container, container_name, *args, **kwargs):
+def remove(container: Container, container_name: str, *args, **kwargs):
```
Comment (alunduil) [inline etest/docker/container.py:53]: Does container contain the container_name?  If so, we can reduce the arity here.

File: etest/tests.py [hunk 2]
```diff
@@ -123,11 +123,11 @@ def run(self):
                 container=container,
             )
 
-            self.failed = is_interrupted or bool(docker.container.wait(container_name))
+            exit_code = docker.container.wait(container_name)["StatusCode"]
```
Comment (alunduil) [inline etest/tests.py:126]: Why give this a name for only one use?

File: etest/tests.py [hunk 3]
```diff
@@ -137,13 +137,14 @@ def run(self):
 
             tag_name = str(self.commands.index(command))
 
+            commit_name = self.name.replace("=", "").replace("[", "").replace("]", "")
```
Comment (alunduil) [inline etest/tests.py:140]: We should move this type of sanitization into the commit function itself.

File: etest/tests.py [hunk 4]
```diff
@@ -137,13 +137,14 @@ def run(self):
...
             docker.container.commit(
-                container_name,
-                repository=self.name,
+                container,
+                repository=commit_name,
                 tag=tag_name,
             )
 
-            image_name = self.name + ":" + tag_name
+            image_name = commit_name + ":" + tag_name
```
Comment (alunduil) [inline etest/tests.py:147]: I'd argue that commit should return the image name but we can figure that out later if that doesn't make sense.

**Review by alunduil — CHANGES_REQUESTED** [review]
Please squash these commits so we have a clear commit set with explanations of changes if history needs to be reviewed.

File: etest/docker/container.py (hunk 1 above)
Comment (LinoBigatti) [reply etest/docker/container.py:53]: Actually, it does. I didnt think about that.

File: etest/tests.py (hunk 2 above)
Comment (LinoBigatti) [reply etest/tests.py:126]: This is a debugging thing that managed to sneak in.

File: etest/tests.py (hunk 3 above)
Comment (LinoBigatti) [reply etest/tests.py:140]: That would only work if commit returns the name of the image, as youve said.

File: etest/tests.py (hunk 2 above)
Comment (alunduil) [reply etest/tests.py]: Could just codify it with a debug logging statement. ;)

File: etest/docker/container.py [hunk 5]
```diff
@@ -6,15 +6,21 @@
...
+from docker.models.containers import Container
+
 from etest.docker import common
 
 CONTAINERS: List = []
 CREATE = True
 
 
-def commit(*args, **kwargs):
+def commit(container: Container, tag: str, repo_: str, *args, **kwargs):
```
Comment (alunduil) [inline etest/docker/container.py:17]: The interface should have the non-mangled name for a better consumer API.  Let's change repo_ to repository here.

**Review by alunduil — CHANGES_REQUESTED** [review]
One last fixup and a squash.  Then I think this is ready to land.

File: etest/docker/container.py (hunk 5 above)
Comment (LinoBigatti) [reply etest/docker/container.py:17]: Sure, I added that to differentiate it from the local scoped repo

**Comment by LinoBigatti** [comment]
Ive set gh settings to auto squash merge so a squash shouldnt be necessary

**Comment by alunduil** [comment]
> Its enabled on a per-pull request basis. You have to manually enable it and set the merge type for each pull request.

Oh interesting.  I can live with that.

**Review by alunduil — CHANGES_REQUESTED** [review]
I'm surprised pre-commit didn't catch that mismatch.

File: etest/tests.py [hunk 6]
```diff
@@ -137,16 +136,15 @@ def run(self):
 
             tag_name = str(self.commands.index(command))
 
-            docker.container.commit(
-                container_name,
-                repository=self.name,
+            image_name = docker.container.commit(
+                container=container,
+                repo_=self.name,
```
[the remark this answers is not in the document]
Comment (LinoBigatti) [reply etest/tests.py:141]: the one time I decide to not test it...

**Review by alunduil — APPROVED** [review]

**Pull request merged**